generated/datasources/transform/decode/role_name_test.go

```diff
@@ -0,0 +1,124 @@
+package decode
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/terraform-providers/terraform-provider-vault/generated/resources/transform/role"
+	"github.com/terraform-providers/terraform-provider-vault/generated/resources/transform/transformation"
+	"github.com/terraform-providers/terraform-provider-vault/schema"
+	"github.com/terraform-providers/terraform-provider-vault/util"
+	"github.com/terraform-providers/terraform-provider-vault/vault"
+)
+
+var roleNameTestProvider = func() *schema.Provider {
+	p := schema.NewProvider(vault.Provider())
+	p.RegisterResource("vault_mount", vault.MountResource())
+	p.RegisterResource("vault_transform_transformation_name", transformation.NameResource())
+	p.RegisterResource("vault_transform_role_name", role.NameResource())
+	p.RegisterDataSource("vault_transform_decode_role_name", RoleNameDataSource())
+	return p
+}()
+
+func TestDecodeBasic(t *testing.T) {
+	isEnterprise := os.Getenv("TF_ACC_ENTERPRISE")
+	if isEnterprise == "" {
+		t.Skip("TF_ACC_ENTERPRISE is not set, test is applicable only for Enterprise version of Vault")
+	}
```

Comment:
This could be extracted to a function:
```go
func SkipIfNotEnterprise(t *testing.T) {
	t.Helper()
	isEnterprise := os.Getenv("TF_ACC_ENTERPRISE")
	if isEnterprise == "" {
		t.Skip("TF_ACC_ENTERPRISE is not set, test is applicable only for Enterprise version of Vault")
	}
}
```

Reply:
Ah! Good idea! https://github.com/terraform-providers/terraform-provider-vault/pull/749/commits/8b3e3927a8532c525c932692981269112571c55c